src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_SymbolFinder.cs

```diff
@@ -55,21 +55,21 @@ internal partial class CodeAnalysisService : IRemoteSymbolFinder
                 var result = await DeclarationFinder.FindAllDeclarationsWithNormalQueryInCurrentProcessAsync(
                     project, query, criteria, this.CancellationToken).ConfigureAwait(false);
 
-                return result.Select(SerializableSymbolAndProjectId.Dehydrate).ToArray();
+                return result.SelectAsArray(SerializableSymbolAndProjectId.Dehydrate);
```

Comment:
I'm not a fan of caching delegates in non-hot paths.   For a given feature, this will be something like 1 additional allocation.  I'm totally fine with that, and with us keeping code simple and clear up until the point that we need to (because it's in a hot path).